src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -39,11 +41,26 @@ impl CommandExt for FlashbackToVersionReadPhase {
     }
 }
 
-pub const FLASHBACK_BATCH_SIZE: usize = 256;
+pub const FLASHBACK_BATCH_SIZE: usize = 256 + 1 /* To store the next key for multiple batches */;
 
+/// FlashbackToVersion contains two phases:
+///   1. Read phase:
+///     - Scan all locks to delete them all later.
+///     - Scan all the latest writes to flashback them all later.
+///   2. Write phase:
+///     - Delete all locks we scanned at the read phase.
+///     - Write the old MVCC version writes for the keys we scanned at the read
+///       phase.
 impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
     fn process_read(self, snapshot: S, statistics: &mut Statistics) -> Result<ProcessResult> {
+        if self.commit_ts <= self.start_ts {
+            return Err(Error::from(ErrorInner::InvalidTxnTso {
+                start_ts: self.start_ts,
+                commit_ts: self.commit_ts,
+            }));
+        }
         let mut reader = MvccReader::new_with_ctx(snapshot, Some(ScanMode::Forward), &self.ctx);
+        // TODO: maybe we should resolve all locks before starting a flashback.
```

Comment:
I think we should. Or delete all the versions between [min-resolve-ts, latest-ts) to provide a transaction conisistency.

Reply:
It's hard to make transaction conisistency during flashback cluster. Because the commit-ts for each region is different and we can't rollback flashbacked region when other region meets error during flashback.

Maybe in TiDB we could ban snapshot read with [min-resolve-ts, flashback-end-ts) time range.